zend_abstract_interface/env/env.h

```diff
@@ -0,0 +1,35 @@
+#ifndef ZAI_ENV_H
+#define ZAI_ENV_H
+
+#include <stddef.h>
+
+typedef enum {
+    /* The buffer 'buf' was successfully written to with the value of the target
+     * environment variable 'name'.
+     */
+    ZAI_ENV_SUCCESS,
+    /* The function is being called outside of a request context. */
+    ZAI_ENV_NOT_READY,
+    /* The environment variable is not set. */
+    ZAI_ENV_NOT_SET,
+    /* The buffer is not large enough to accommodate the length of the value. */
+    ZAI_ENV_BUFFER_TOO_SMALL,
+    /* API usage error. */
+    ZAI_ENV_ERROR,
+} zai_env_result;
+
+/* Fills 'buf' with the value of a target environment variable identified by
+ * 'name'. Must be called within a request context. If the active SAPI has a
+ * custom environment variable handler, the SAPI handler is used to access the
+ * environment variable. If there is no custom handler, the environment variable
+ * is accessed from the host using getenv().
+ *
+ * For error conditions, a return value other than ZAI_ENV_SUCCESS is returned
+ * and 'buf' is made an empty string. If the buffer size 'buf_size' is not big
+ * enough to contain the value, ZAI_ENV_BUFFER_TOO_SMALL will be returned and
+ * 'buf' will be an empty string; e.g. this API does not attempt to truncate
+ * the value to accommodate the buffer size.
```

Comment:
Regarding the buffer size, should we maybe define a default buffer size (`#define ZAI_ENV_MAX_BUFSIZ`) to 32768? which we can then use for our "unbounded" use cases, like the array map etc. (at least on linux getenv() will never return a string longer than ~32760 currently, so that seems to me like a sane upper limit).

Then usages of zai_getenv() will do `char envbuf[ZAI_ENV_MAX_BUFSIZ]` - and when we ever notice we need bigger buffers, we can increase that centrally here.

Reply:
Excellent suggestion. I'll implement this.

Reply:
> Then usages of zai_getenv() will do char envbuf[ZAI_ENV_MAX_BUFSIZ] - and when we ever notice we need bigger buffers, we can increase that centrally here.

I don't think it's smart to increase that number if we're putting it on the stack. For what it's worth, PHP's `ALLOCA_FLAG` does [32KiB stack allocations](https://heap.space/xref/php-src/Zend/zend_portability.h?r=c276c16b#349) and anything larger will go to the heap. What do you think?

Reply:
While that's right, at that point, we can grep the usages of this `ZAI_ENV_MAX_BUFSIZ` and replace the usages by a DO_ALLOCA(). But that currently should not be needed.